src/sicks3000.cc

```diff
@@ -310,3 +310,330 @@ unsigned short SickS3000::CreateCRC(uint8_t *Data, ssize_t length)
+  "Check whether the S3000 is operated as per the permissible ambient conditions.",
+  "Verify that the machine-side EDM is connected correctly.",
+  "Disconnect the connection to the host device (see section 6.1.1 on page 97). Check the connected device and the connection to this device.",
+  "Check the wiring for cross-circuits. Or: Check whether the reset pulse complies with the requirements (see Fig. 95 on page 143).",
+  "Check the path of the vehicle. Or: Check the operating process of the monitored machine or system. If necessary, check the configuration of the monitoring cases with the aid of the CDS.",
+  "Check the path of the vehicle. Or: Check the operating process of the monitored machine or system. If necessary, check the configuration of the monitoring cases with the aid of the CDS.",
+  "Check the operation of the digital control inputs.",
+  "Check the wiring, the digital control inputs or the wiring to the devices connected via EFI.",
+  "Check the wiring, the digital control inputs or the wiring to the devices connected via EFI.",
+  "Check the wiring, the digital control inputs or the wiring to the devices connected via EFI.",
+  "Check the wiring, the digital control inputs or the wiring to the devices connected via EFI.",
+  "Check the wiring, the digital control inputs or the wiring to the devices connected via EFI.",
+  "No error. Readiness is re-established by switching to a different monitoring case or withdrawing the stand-by bit via EFI.",
+  "Carry out a fault diagnosis of the device connected with the S3000.",
+  "No error",
+  "Clean the front screen.",
+  "Clean the front screen.",
+  "Clean the front screen.",
+  "Clean the front screen.",
+  "Clean the front screen.",
+  "Clean the front screen.",
+  "Re-fit the new front screen (then perform front screen calibration). If at the time of the error a front screen was fitted: Check whether the S3000 is being dazzled by an external light source, e.g. headlight, infrared light source, stroboscopic light, sun etc.",
+  "Carry out a front screen calibration or replace the S3000, if necessary.",
+  "Carry out a front screen calibration or replace the S3000, if necessary.",
+  "Replace the S3000.",
+  "Replace the I/O module.",
+  "Check whether the correct I/O module has been used, and replace if necessary."
+};
+
+#define MAX_READ_ATTEMPTS 10
+int SickS3000::CheckOperatingDataBlock(int retry_count)
+{
+  if (serial!=NULL)
+  {
+    char interrupt_byte[1] = {0x41};
+    char get_token[20] = {0x00,0x00,0x41,0x44,0x19,0x00,0x00,0x05,0xFF,0x07,0x19,0x00,0x00,0x05,0xFF,0x07,0x07,0x0F,0x9F,0xD0};
+    char request_block_11[10] = {0x00,0x00,0x45,0x44,0x0B,0x00,0x00,0x56,0xFF,0x07};
+    char release_token[20] = {0x00,0x00,0x41,0x44,0x19,0x00,0x00,0x05,0xFF,0x07,0x19,0x00,0x00,0x05,0xFF,0x07,0x00,0x00,0xE7,0xB8};
```

Comment:
They are the commands sent to the S3000, and they come from that document you linked in the JIRA.  I'll add some comments explaining this.